web/src/app/WebApp/Installers/BaseSetup.php

```diff
@@ -49,7 +57,18 @@ public function getDocRoot($append_relative_path=null) : string
         if(empty($domain_path) || ! is_dir($domain_path)) {
             throw new \Exception("Error finding domain folder ($domain_path)");
         }
-        return Util::join_paths($domain_path, "public_html", $append_relative_path);
+        $subDir = $this->getAppDirInstall()?$this->getAppDirInstall()!='/'?$this->getAppDirInstall():'':'';
+        if ($subDir==''){
+                return Util::join_paths($domain_path, "public_html", $append_relative_path);
+        }
+
+        $directory = Util::join_paths($domain_path, "public_html", $subDir);
+
+        if (!file_exists($directory)) {
+                mkdir($directory, 0755, true);
```

Comment:
Create directory with https://github.com/hestiacp/hestiacp/blob/main/bin/v-add-fs-directory

This will make sure the folder has the correct permissions

Reply:
In hestiacp there Is a PHP class that do this (call v-add-fs-directory)??? i ask so as not to make redundant functions

Reply:
https://github.com/hestiacp/hestiacp/blob/12b39f9435538a19923d53a00b757eea897a4d58/web/src/app/WebApp/Installers/Opencart/OpencartSetup.php#L71

This example is for v-delete-fs-directory but can also be used for v-add-fs-directory

Reply:
So it will be:
$this->appcontext->runUser('v-add-fs-directory', [$this->getDocRoot("$directory")